src/node_worker.cc

```diff
@@ -816,6 +816,114 @@ void Worker::Unref(const FunctionCallbackInfo<Value>& args) {
+    // object.
+
+    env->SetImmediateThreadsafe(
+        [taker = std::move(taker),
+         heap_stats = std::move(heap_stats)](Environment* env) mutable {
+          Isolate* isolate = env->isolate();
+          HandleScope handle_scope(isolate);
+          Context::Scope context_scope(env->context());
+
+          AsyncHooks::DefaultTriggerAsyncIdScope trigger_id_scope(taker->get());
+
+          Local<v8::Name> heap_stats_names[] = {
+              FIXED_ONE_BYTE_STRING(isolate, "total_heap_size"),
+              FIXED_ONE_BYTE_STRING(isolate, "total_heap_size_executable"),
+              FIXED_ONE_BYTE_STRING(isolate, "total_physical_size"),
+              FIXED_ONE_BYTE_STRING(isolate, "total_available_size"),
+              FIXED_ONE_BYTE_STRING(isolate, "used_heap_size"),
+              FIXED_ONE_BYTE_STRING(isolate, "heap_size_limit"),
+              FIXED_ONE_BYTE_STRING(isolate, "malloced_memory"),
+              FIXED_ONE_BYTE_STRING(isolate, "peak_malloced_memory"),
+              FIXED_ONE_BYTE_STRING(isolate, "does_zap_garbage"),
+              FIXED_ONE_BYTE_STRING(isolate, "number_of_native_contexts"),
+              FIXED_ONE_BYTE_STRING(isolate, "number_of_detached_contexts"),
+              FIXED_ONE_BYTE_STRING(isolate, "total_global_handles_size"),
+              FIXED_ONE_BYTE_STRING(isolate, "used_global_handles_size"),
+              FIXED_ONE_BYTE_STRING(isolate, "external_memory")};
+
+          // Define an array of property values
+          Local<Value> heap_stats_values[] = {
+              Number::New(isolate, heap_stats.total_heap_size()),
+              Number::New(isolate, heap_stats.total_heap_size_executable()),
+              Number::New(isolate, heap_stats.total_physical_size()),
+              Number::New(isolate, heap_stats.total_available_size()),
+              Number::New(isolate, heap_stats.used_heap_size()),
+              Number::New(isolate, heap_stats.heap_size_limit()),
+              Number::New(isolate, heap_stats.malloced_memory()),
+              Number::New(isolate, heap_stats.peak_malloced_memory()),
+              Boolean::New(isolate, heap_stats.does_zap_garbage()),
+              Number::New(isolate, heap_stats.number_of_native_contexts()),
+              Number::New(isolate, heap_stats.number_of_detached_contexts()),
+              Number::New(isolate, heap_stats.total_global_handles_size()),
+              Number::New(isolate, heap_stats.used_global_handles_size()),
+              Number::New(isolate, heap_stats.external_memory())};
+
+          // Create the object with the property names and values
+          Local<Object> stats = Object::New(isolate,
+                                            Null(isolate),
+                                            heap_stats_names,
+                                            heap_stats_values,
+                                            arraysize(heap_stats_names));
```

Comment:
A `DCHECK()` that ensures that names and values array have the same size would be perfect for future reference.

Reply:
Done